File: Week -2/Qs5.c
#include <stdio.h>

int main() {
    int arr[100], n, i;
    int max, min;
    int max_pos, min_pos;

    printf("Enter number of elements in the array: ");
    scanf("%d", &n);

    printf("Enter %d elements: ", n);
    for(i = 0; i < n; i++) {
        scanf("%d", &arr[i]);
    }

    max = min = arr[0];
    max_pos = min_pos = 1;

    for(i = 1; i < n; i++) {
        if(arr[i] > max) {
            max = arr[i];
            max_pos = i + 1;
        }
        if(arr[i] < min) {
            min = arr[i];
            min_pos = i + 1;
        }
    }
    printf("Maximum number: %d\n", max);
    printf("Position of maximum number: %d\n", max_pos);
    printf("Minimum number: %d\n", min);
    printf("Position of minimum number: %d\n", min_pos);

    return 0;
}

File: Week -2/Qs5.py
n = int(input("Enter number of elements in the array: "))
arr = []
print(f"Enter {n} elements: ")
for i in range(n):
    element = int(input())
    arr.append(element)
max_value = min_value = arr[0]
max_pos = min_pos = 0  # Positions are 0-based initially
for i in range(1, n):
    if arr[i] > max_value:
        max_value = arr[i]
        max_pos = i
    if arr[i] < min_value:
        min_value = arr[i]
        min_pos = i
print(f"Maximum number: {max_value}")
print(f"Position of maximum number: {max_pos + 1}")  # Convert to 1-based index
print(f"Minimum number: {min_value}")
print(f"Position of minimum number: {min_pos + 1}")  # Convert to 1-based index
